crates/core/src/processor.rs

```diff
@@ -0,0 +1,158 @@
+//! Processor is reponsible for running the module.
+//! This run is sound execution of the module.
+//! This will be most abstract layer of the processor.
+
+use alloy_dyn_abi::DynSolValue;
+use alloy_merkle_tree::standard_binary_tree::StandardMerkleTree;
+use alloy_primitives::{hex::FromHex, FixedBytes, Keccak256, B256, U256};
+use anyhow::Result;
+use hdp_primitives::processed_types::{cairo_format::AsCairoFormat, v1_query::ProcessedResult};
+use serde::Serialize;
+use std::{path::PathBuf, str::FromStr};
+
+use hdp_provider::evm::{AbstractProvider, AbstractProviderConfig};
+
+use crate::cairo_runner::run::Runner;
+
+pub struct Processor {
+    runner: Runner,
+    _provider: AbstractProvider,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ProcessorResult {
+    /// leaf of result merkle tree
+    pub task_results: Vec<String>,
+    /// leaf of task merkle tree
+    pub task_commitments: Vec<String>,
+    /// tasks inclusion proofs
+    pub task_inclusion_proofs: Vec<Vec<FixedBytes<32>>>,
+    /// results inclusion proofs
+    pub results_inclusion_proofs: Vec<Vec<FixedBytes<32>>>,
+    /// root of the results merkle tree
+    pub results_root: String,
+    /// root of the tasks merkle tree
+    pub tasks_root: String,
+    /// mmr id
+    pub used_mmr_id: u64,
+    /// mmr size
+    pub used_mmr_size: u64,
+}
+
+impl ProcessorResult {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        task_results: Vec<String>,
+        task_commitments: Vec<String>,
+        task_inclusion_proofs: Vec<Vec<FixedBytes<32>>>,
+        results_inclusion_proofs: Vec<Vec<FixedBytes<32>>>,
+        results_root: String,
+        tasks_root: String,
+        used_mmr_id: u64,
+        used_mmr_size: u64,
+    ) -> Self {
+        Self {
+            task_results,
+            task_commitments,
+            task_inclusion_proofs,
+            results_inclusion_proofs,
+            results_root,
+            tasks_root,
+            used_mmr_id,
+            used_mmr_size,
+        }
+    }
+}
+
+impl Processor {
+    pub fn new(provider_config: AbstractProviderConfig, program_path: PathBuf) -> Self {
+        let runner = Runner::new(program_path);
+        let provider = AbstractProvider::new(provider_config);
+        Self {
+            runner,
+            _provider: provider,
+        }
+    }
+
+    pub async fn process(
+        &self,
+        requset: ProcessedResult,
+        pie_path: String,
+    ) -> Result<ProcessorResult> {
```

Comment:
yeah we have double result name here that suggests .unwrap().unwrap() or flattening result